Dockerfile

```diff
@@ -1,4 +1,20 @@
-FROM nixos/nix
-ADD . .
-RUN nix-channel --update
-CMD nix-build default.nix
\ No newline at end of file
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt update && apt install -y curl xz-utils git sudo bash && \
+    git config --global --add safe.directory /workspaces/* && \
+    # Setup and switch to a non-root user (username: nonroot) to install nix.
+    useradd -m nonroot && mkhomedir_helper nonroot && \
+    mkdir -m 0755 /nix && chown -R nonroot /nix && chmod u+wxr ./nix
+
+USER nonroot
+ENV USER=nonroot
+ADD --chown=nonroot:nonroot . /home/nonroot/tmp_src
+RUN curl -L https://nixos.org/nix/install > ~/nix_install.sh && \
+    chmod u+x ~/nix_install.sh && \
+    ~/nix_install.sh --no-daemon && \
+    rm ~/nix_install.sh && \
+    . /home/nonroot/.nix-profile/etc/profile.d/nix.sh && \
+    nix-channel --update && \
+    (cd /home/nonroot/tmp_src && nix-shell --option sandbox false) && \
```

Comment:
Everything is moving to nix-flakes so probably `nix build .#something` (`.#ci`?) would probably be more precise.

Reply:
That's good to know, thanks. I'll see what I can do.